docs/DeveloperGuide.adoc

```diff
@@ -729,8 +729,62 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
 |`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
 
+|`* * *` |user with many people in the address book |list all people in address book |see whose contacts the address book already has
+
+|`* * *` |user |clear all people in the address book |remove all contacts conveniently
+
+|`* * *` |user |list the command history |view all commands already entered
+
+|`* * *` |user |undo previous command |amend any mistakes
+
+|`* * *` |user |redo the previously undone command |amend any mistakes
+
+|`* * *` |user |edit a person's information |update changes to their particulars
+
+|`* * *` |user with many people in the address book |make relation links between contacts |link up contacts that have connections with each other
+
+|`* * *` |user with many people in the address book |sort contacts by different fields in alphabetical order |arrange them according to the fields
+
+|`* * *` |user with many people in the address book |sort contacts to my preference |arrange them according to their importance / significance
+
+|`* * *` |user |find users based on tags |locate different groups easily
+
+|`* * *` |user |rename an existing tag |update a tag which is no longer suitable to describe the people who have them
+
+|`* * *` |user |run the executable file of the address book |access the application conveniently
+
+|`* *` |user with many people in the address book |delete people in address book by tags |remove people of the same group
+
+|`* *` |user with many people in the address book |change the view mode of GUI |display different number of contacts per page
+
+|`* *` |user |change font size, colours, etc |customize the layout according to preference
+
+|`* *` |original user |set password for access |enhance the security of Address Book
+
+|`* *` |user |view [private contact details](#private-contact-detail) |view details of hidden contacts
+
+|`* *` |user with many people in the address book |sort people by birthday |check for upcoming birthdays
+
+|`* *` |user |highlight contacts |place it at the top
+
+|`* *` |user |displays the most searched contacts |display the most searched contacts
+
+|`* *` |user |tag multiple people at the same time |quickly tag all people who share the similarities conveniently
+
 |`* *` |user |hide link:#private-contact-detail[private contact details] by default |minimize chance of someone else seeing them by accident
 
+|`*` | user | update data to server | access it from any device
+
+|`*` | user | add photos to contacts in GUI | recognize the contacts easily
+
+|`*` | user | add speech recognition | make it a hands-free device
+
+|`*` | user | display contacts address on map | access their location
+
+|`*` | user | import data file from another device | access from different devices
+
+|`*` | user | export data file to another format | transfer to different devices
+
```

Comment:
Overall, all user stories follow the correct form. However, you need to pay more attention whether a requirement is a functional/feature or not.